lib/teleterm/services/connectmycomputer/connectmycomputer.go

```diff
@@ -254,3 +257,49 @@ func (c *RoleSetupConfig) CheckAndSetDefaults() error {
 
 	return nil
 }
+
+// CreateNodeToken creates a node join token that is valid for 5 minutes
+func (t *TokenProvisioner) CreateNodeToken(ctx context.Context, provisioner Provisioner, cluster *clusters.Cluster) (types.ProvisionToken, error) {
+	tokenName, err := utils.CryptoRandomHex(auth.TokenLenBytes)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	var req types.ProvisionTokenSpecV2
+	req.Roles = types.SystemRoles{types.RoleNode}
+	req.SuggestedLabels = types.Labels{
+		types.ConnectMyComputerNodeOwnerLabel: apiutils.Strings{cluster.GetLoggedInUser().Name},
+	}
+	expires := time.Now().UTC().Add(5 * time.Minute)
+
+	provisionToken, err := types.NewProvisionTokenFromSpec(tokenName, expires, req)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	err = provisioner.CreateToken(ctx, provisionToken)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return provisionToken, nil
+}
+
+// DeleteToken deletes a join token
+func (t *TokenProvisioner) DeleteToken(ctx context.Context, provisioner Provisioner, token string) error {
+	err := provisioner.DeleteToken(ctx, token)
+	return trace.Wrap(err)
+}
+
+// Provisioner represents [services.Provisioner] methods used by [TokenProvisioner].
+// During a normal operation, [auth.ClientI] is passed as this interface.
+type Provisioner interface {
+	// See [services.Provisioner.CreateToken].
+	CreateToken(ctx context.Context, token types.ProvisionToken) error
+	// See [services.Provisioner.DeleteToken].
+	DeleteToken(ctx context.Context, token string) error
+}
+
+type TokenProvisioner struct {
```

Comment:
From what I've seen in std lib of Go (net URL, http Server), structs defined by the package are typically defined before methods for them are written out. Let's use this convention here, especially because `RoleSetup` already follows it.

It also makes the code a bit easier to read, because now we have

```go
func (c *RoleSetupConfig) CheckAndSetDefaults()
```

followed immediately by

```go
func (t *TokenProvisioner) CreateNodeToken
```

Reply:
Done!